### PR TITLE
AL-5635: Create library for adapted signer

### PR DESCRIPTION
**Pull request by maccelf**

### CODE REVIEW

File: src/signer/base_signer.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    safe_mkdir,
+)
+from signer.utils.pgp_utils import init_gpg
+
+
+class SignStatusEnum(enum.IntEnum):
+    SUCCESS = 1
+    READ_ERROR = 2
+    NO_SIGNATURE = 3
+    WRONG_SIGNATURE = 4
```
Comment (anfimovdm) [inline src/signer/base_signer.py:36]: Please move it into constants.py

File: src/signer/base_signer.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+
+
+class BaseSigner(object):
```
Comment (anfimovdm) [inline src/signer/base_signer.py:39]: No need to inherit from object here

File: src/signer/base_signer.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+    def _check_signature(self, files, key_id):
+        errors = []
+        key_id_lower = key_id.lower()
+        ts = rpm.TransactionSet()
+        ts.setVSFlags(rpm._RPMVSF_NOSIGNATURES)
+        subkeys = [i.lower() for i in self.__password_db.get_subkeys(key_id)]
+
+        def check(pkg_path: str) -> typing.Tuple[SignStatusEnum, str]:
+            if not os.path.exists(pkg_path):
```
Comment (anfimovdm) [inline src/signer/base_signer.py:58]: We can use pathlib module instead

File: src/signer/base_signer.py [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+        raise NotImplementedError('Needs to be implemented in child classes')
+
+    def _check_signature(self, files, key_id):
+        errors = []
+        key_id_lower = key_id.lower()
+        ts = rpm.TransactionSet()
+        ts.setVSFlags(rpm._RPMVSF_NOSIGNATURES)
+        subkeys = [i.lower() for i in self.__password_db.get_subkeys(key_id)]
+
+        def check(pkg_path: str) -> typing.Tuple[SignStatusEnum, str]:
```
Comment (anfimovdm) [inline src/signer/base_signer.py:57]: It could be a staticmethod

File: src/signer/base_signer.py [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+                    return SignStatusEnum.SUCCESS, ''
+                if subkeys and sig in subkeys:
+                    return SignStatusEnum.SUCCESS, ''
+
+            return SignStatusEnum.WRONG_SIGNATURE, sig
+
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            futures = {}
+            for file_ in files:
+                futures[executor.submit(check, file_)] = file_
```
Comment (anfimovdm) [inline src/signer/base_signer.py:83]: Could be a dict comprehension

File: src/signer/base_signer.py [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+
+        # Detect if there are some RPMs in the payload
+        for package in task['packages']:
+            package_type = package.get('type', 'rpm')
+            if package_type == 'rpm':
+                has_rpms = True
+                break
+
+        try:
+            with ThreadPoolExecutor(max_workers=4) as executor:
```
Comment (anfimovdm) [inline src/signer/base_signer.py:152]: Please move that part in try-except block into a separate func

File: src/signer/errors.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        exit_code : int
+            Command exit code.
+        stdout : str
+            Command stdout.
+        stderr : str
+            Command stderr.
+        command : list, optional
+            Executed command.
+        """
+        super(CommandExecutionError, self).__init__(message)
```
Comment (anfimovdm) [inline src/signer/errors.py:73]: No need to pass arguments to super method

File: src/signer/package_sign.py [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+            '\nTraceback: {3}'.format(
+                ar_path, proc.returncode, err, traceback.format_exc()
+            )
+        )
+        logging.error(msg)
+        raise PackageSignError(msg)
+    return file_names.split()
+
+
+def _unpack_ar_file(ar_path, file_name, stdout):
```
[the remark this answers is not in the document]
Comment (anfimovdm) [reply src/signer/package_sign.py:62]: I think we should discuss it first with our release managers, maybe we don't need to sign deb-like packages at all

**Review by anfimovdm — CHANGES_REQUESTED** [review]
Also, please add type annotations

**Comment by maccelf** [comment]
Closed because of https://github.com/AlmaLinux/build-system-common-library/pull/1

**Pull request closed without merging**